File: app/src/main/java/com/parabola/newtone/mvp/view/TrackAdditionalInfoView.java
package com.parabola.newtone.mvp.view;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;
import com.parabola.domain.model.Track;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface TrackAdditionalInfoView extends MvpView {
    void setTrack(Track track);

    @StateStrategyType(OneExecutionStateStrategy.class)
    void closeScreen();
}

File: app/src/main/java/com/parabola/newtone/mvp/view/TrackAdditionalInfoView.kt
package com.parabola.newtone.mvp.view

import com.parabola.domain.model.Track
import moxy.MvpView
import moxy.viewstate.strategy.alias.AddToEndSingle
import moxy.viewstate.strategy.alias.OneExecution

@AddToEndSingle
interface TrackAdditionalInfoView : MvpView {
    fun setTrack(track: Track)

    @OneExecution
    fun closeScreen()
}